tests/interfaces/network.go

```diff
@@ -0,0 +1,52 @@
+package interfaces
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"math/big"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ava-labs/subnet-evm/ethclient"
+	teleportermessenger "github.com/ava-labs/teleporter/abi-bindings/go/Teleporter/TeleporterMessenger"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type SubnetTestInfo struct {
+	SubnetID                  ids.ID
+	BlockchainID              ids.ID
+	NodeURIs                  []string
+	WSClient                  ethclient.Client
+	RPCClient                 ethclient.Client
+	EVMChainID                *big.Int
+	TeleporterRegistryAddress common.Address
+	TeleporterMessenger       *teleportermessenger.TeleporterMessenger
+}
+
+// Defines the interface for the network setup functions used in the E2E tests
+type Network interface {
+	// Returns all of the subnets support by this network.
+	GetSubnetsInfo() []SubnetTestInfo
+
+	// Returns the Teleporter contract address for all subnets in this network.
+	GetTeleporterContractAddress() common.Address
+
+	// An address and corresponding key that has native tokens on each of the subnets in this network.
+	GetFundedAccountInfo() (common.Address, *ecdsa.PrivateKey)
+
+	// Whether or not the funded wallet is capable of relaying messages between subnets in this network.
+	// Intended to be true for local networks where all nodes are querable by the test application for their
+	// BLS signatures, and false for testnet networks where test application does not neccarily have
+	// connections with each validator.
+	SupportsIndependentRelaying() bool
+
+	// For implementations where SupportsIndependentRelaying() is true, relays the specified message between the
+	// two subnets,and returns the receipt of the transaction the message was delivered in.
+	// For implementations where SupportsIndependentRelaying() is false, waits for the specific message to be relayed
+	// by an external relayer, and returns the receipt of the transaction the message was delivered in.
+	RelayMessage(ctx context.Context,
+		sourceReceipt *types.Receipt,
+		source SubnetTestInfo,
+		destination SubnetTestInfo,
+		expectSuccess bool) *types.Receipt
+}
```

Comment:
I'm wondering if it would be worth having a separate `Network` implementation for `deliver_to_wrong_chain.go` that implements `RelayMessage` in that test's unique way for Fuji.  That way, we could remove `SupportsIndependentRelaying` altogether, rather than have two tightly coupled methods.

Reply:
I think it'd be odd to have a `Network` implementation for a specific test, rather than keeping it one per "network type" (i.e. local network run by the tests themselves, or independent live network).

I don't love having two coupled methods, but think it's reasonable it's reasonable in this case, especially since one is just a boolean flag.

Reply:
Yeah, I agree with that. Better to have coupled methods within the same interface than to have tests depend on specific interface implementations (with the notable exception that certain tests only run on local networks)